lib/cli-update-connection.ts

```diff
@@ -61,26 +69,26 @@ export default function CliUpdateConnection() {
             connectionString: settings.connectionString
         };
 
+        cli.print('Updating connection string...');
         client.datasets.setAllConnections(settings.collection, settings.workspace, settings.dataset, params, (err, result) => {
-            if (err) {
-                cli.error('Error updating connection string');
-                cli.error(err);
-                return;
+            if (err && typeof (callback) === 'function') {
+                callback(err);
             }
 
             cli.success('Connection string successfully updated');
             cli.print('Dataset: ', settings.dataset);
             cli.print('ConnectionString: ', settings.connectionString);
 
-            if (callback) {
+            if (result && typeof (callback) === 'function') {
                 callback(null, result);
             }
         });
     }
 
     function updateCredentials(client: powerbi.PowerBIClient, settings: any, callback?): void {
+        cli.print('Getting gateway datasources...');
         client.datasets.getGatewayDatasources(settings.collection, settings.workspace, settings.dataset, (err, result) => {
-            if (err) {
+            if (err && typeof (callback) === 'function') {
                 return callback(err);
```

Comment:
I think this changes the logic.
Previously, if there was an error it was guaranteed to return. Now it will only return if there is also a callback. 
I think you intended to only execute the call back if it is provided but alwasy return to stop further execution.

Also, for these, is there nothing else we can do when there is an error but no callback?  Maybe we can at least provided a default callback which will log the error instead of swallowing?